File: mp-b/admin/api/accident.ts
/**
 * 事故登记相关接口
 * 对应后端 AccidentController（/accident/...）
 */
import { get, post } from '../utils/request'
import { PageResult, mockPage } from './elder'

const PAGE_NUM = 1
const PAGE_SIZE = 200

export interface AccidentItem {
	id: number
	elder_name: string
	staff_name: string
	occur_date: string
}

export interface AccidentDetail {
	id: number
	elder_name: string
	staff_id: number
	occur_date: string
	description: string
	picture: string
}

export interface AccidentForm {
	id?: number
	elder_id: number
	staff_id: number
	occur_date: string
	description: string
	picture: string
}

// ====== MOCK 数据（后台未完成，测试前端用；后台完成后放开各函数内注释并删除本段）======
const mockAccidentList: AccidentItem[] = [
	{ id: 1, elder_name: '李德福', staff_name: '孙护工', occur_date: '2026-08-15 14:00' }
]
const mockAccidentDetail: AccidentDetail = { id: 1, elder_name: '李德福', staff_id: 3, occur_date: '2026-08-15 14:00', description: '散步时轻微摔倒，已及时处理并包扎', picture: '' }

/** 事故列表（全量） */
export function pageAccidentByKey(elderName?: string, staffName?: string): Promise<PageResult<AccidentItem>> {
	// return get<PageResult<AccidentItem>>('/accident/pageAccidentByKey', {
	// 	page_num: PAGE_NUM,
	// 	page_size: PAGE_SIZE,
	// 	elder_name: elderName || undefined,
	// 	staff_name: staffName || undefined
	// })
	// ====== MOCK（后台完成并放开上方注释后，可删除此段）======
	let list = mockAccidentList
	if (elderName) list = list.filter((a) => a.elder_name.indexOf(elderName) >= 0)
	if (staffName) list = list.filter((a) => a.staff_name.indexOf(staffName) >= 0)
	return Promise.resolve(mockPage(list))
}

export function getAccidentById(id: number): Promise<AccidentDetail> {
	// return get<AccidentDetail>('/accident/getAccidentById', { id })
	// ====== MOCK（后台完成并放开上方注释后，可删除此段）======
	return Promise.resolve({ ...mockAccidentDetail, id })
}

export function addAccident(form: AccidentForm): Promise<void> {
	// return post<void>('/accident/addAccident', form)
	// ====== MOCK（后台完成并放开上方注释后，可删除此段）======
	return Promise.resolve()
}

export function editAccident(form: AccidentForm): Promise<void> {
	// return post<void>('/accident/editAccident', form)
	// ====== MOCK（后台完成并放开上方注释后，可删除此段）======
	return Promise.resolve()
}

export function deleteAccident(id: number): Promise<void> {
	// return post<void>('/accident/deleteAccident', { id })
	// ====== MOCK（后台完成并放开上方注释后，可删除此段）======
	return Promise.resolve()
}

File: backend/internal/model/dto/accident.go
package dto

import "time"

// ============ AccidentController 请求 ============

// PageAccidentByKeyReq 分页查询事故登记请求
// @request
type PageAccidentByKeyReq struct {
	PageNum   *int    `json:"page_num" valid:"required"`  // 页码
	PageSize  *int    `json:"page_size" valid:"required"` // 条数
	ElderName *string `json:"elder_name"`                 // 老人姓名
	StaffName *string `json:"staff_name"`                 // 护工姓名
}

// EditAccidentReq 编辑事故登记请求
// @request
type EditAccidentReq struct {
	ID *int64 `json:"id" valid:"required"` // id
	AccidentQuery
}

// AddAccidentReq 新增事故登记请求（继承 EditAccidentReq）
// @request
type AddAccidentReq struct {
	AccidentQuery
	ElderID *int64 `json:"elder_id" valid:"required"` // 老人编号
}

type AccidentQuery struct {
	StaffID     *int64     `json:"staff_id" valid:"required"`    // 护工编号
	OccurDate   *time.Time `json:"occur_date" valid:"required"`  // 发生时间
	Description *string    `json:"description" valid:"required"` // 事故描述
	Picture     *string    `json:"picture" valid:"required"`     // 事故图片
}

// ============ AccidentController 响应 ============

// PageAccidentByKeyResp 分页查询事故登记响应
// @response
type PageAccidentByKeyResp struct {
	ID        int64     `json:"id"`         // id
	ElderName string    `json:"elder_name"` // 老人姓名
	StaffName string    `json:"staff_name"` // 护工姓名
	OccurDate time.Time `json:"occur_date"` // 发生时间
}

// GetAccidentByIDResp 根据编号获取事故登记响应
// @response
type GetAccidentByIDResp struct {
	ID          int64     `json:"id"`          // id
	ElderName   string    `json:"elder_name"`  // 老人姓名
	StaffID     int64     `json:"staff_id"`    // 护工编号
	OccurDate   time.Time `json:"occur_date"`  // 发生时间
	Description string    `json:"description"` // 事故描述
	Picture     string    `json:"picture"`     // 事故图片
}
